frontend/src/services/objectService.ts

```diff
@@ -14,10 +14,40 @@ export default {
    * @param {AxiosRequestConfig} axiosOptions Axios request config options
    * @returns {Promise} An axios response
    */
-  createObject(object: any, bucketId?: string, axiosOptions?: AxiosRequestConfig) {
+  createObject(
+    object: any,
+    headers: {
+      metadata?: Array<{ key: string; value: string }>,
+    },
+    params: {
+      bucketId?: string,
+      tagset?: Array<{ key: string; value: string }>
+    },
+    axiosOptions?: AxiosRequestConfig) {
+
+    // Map the metadata if required
+    let metadata;
+    if (headers.metadata) {
+      metadata = Object.assign({},
+        ...(headers.metadata.map((x: { key: string; value: string }) => ({ [x.key]: x.value })))
+      );
+    }
+
+    // Map the tagset if required
+    let tagset;
+    if (params.tagset) {
+      tagset = Object.assign({}, ...(params.tagset.map((x: { key: string; value: string }) => ({ [x.key]: x.value }))));
+    }
+
     const config = {
-      headers: { 'Content-Type': 'multipart/form-data' },
-      params: { bucketId: bucketId },
+      headers: {
+        'Content-Type': 'multipart/form-data',
+        ...metadata
+      },
+      params: {
+        bucketId: params.bucketId,
+        tagset: tagset
+      }
     };
```

Comment:
With a bit of clever reordering, you can avoid reassignment hell by reutilizing the config object and reassigning attributes in those values instead. Something like the following would be slightly more efficient and concise:
```ts
const config = {
  headers: { 'Content-Type': 'multipart/form-data' },
  params: { bucketId: bucketId },
};

// Map the metadata if required
if (headers.metadata) {
  config.headers = {
    ...config.headers,
    ...Object.fromEntries((headers.metadata.map((x: { key: string; value: string }) => ([x.key, x.value ]))))
  };
}

// Map the tagset if required
if (params.tagset) {
  config.params.tagset = Object.fromEntries((params.tagset.map((x: { key: string; value: string }) => ([x.key, x.value]))));
}
```